skbio/sequence/tests/test_sequence.py

```diff
@@ -1156,6 +1156,43 @@ def test_count_on_subclass(self):
         self.assertIn("Sequence", str(cm.exception))
         self.assertIn("SequenceSubclass", str(cm.exception))
 
```

Comment:
Can you add these tests?

- [ ] calling `replace` on a sequence without metadata is handled efficiently (check that `self` and the returned sequence have `None` for `_metadata` and `_positional_metadata` attributes)
- [ ] calling `replace` doesn't mutate the original sequence (not easy to do but still good to test)
- [ ] passing `where` as an iterable of ints and/or slices
- [ ] passing `where` as a `str` to `positional_metadata`
- [ ] passing `character` that isn't a single character string raises an error
- [ ] passing `character` that is a unicode character outside of ASCII range (0-255) raises an error

One way to accomplish the last test is:

```python
if type(character) is not bytes:
    character = character.encode('ascii')
```